frontend/add-leads/add-event-step.component.tsx

```diff
@@ -106,12 +109,28 @@ export default function AddEventStep(props: AddEventStepProps) {
 
   function existingEventInputs() {
     return (
-      <div>
-        <label htmlFor="existing-events-list">Choose event:</label>
-        <select id="existing-events-list">
-          <option value="To-DO">Still need to implement this</option>
-        </select>
-      </div>
+      <>
+        <div>
+          <label htmlFor="existing-events-list">Choose event:</label>
+          <select
+            id="existing-events-list"
+            value={existingEventId}
+            onChange={evt => setExistingEventId(Number(evt.target.value))}
+          >
+            <option value="">(Select an event)</option>
+            {allEvents.map(event => (
+              <option key={event.id} value={event.id}>
+                {event.eventName} ({event.eventDate})
+              </option>
+            ))}
+          </select>
+        </div>
+        <div className="actions">
+          <button className="primary" type="submit" disabled={!existingEventId}>
+            Reuse event
```

Comment:
Just a wording suggestion. Instead of "Reuse event" would "Add to existing event" convey it better on their end?